pyopenms_viz/_matplotlib/core.py

```diff
@@ -732,7 +757,6 @@ def _add_box_boundaries(self, annotation_data):
                 loc=matplotlibLegendLoc,
                 title=self.feature_config.legend.title,
                 prop={"size": self.feature_config.legend.fontsize},
-                bbox_to_anchor=self.feature_config.legend.bbox_to_anchor,
```

Comment:
why is this line removed? I think this is needed for 2D legend placement

Reply:
Sorry - I forgot to mention this. There seems to be a weird interaction between `bbox_to_anchor` and `tight_layout()`, where the plot gets resized when the legend is added. With the legend enabled not only the plot area shrinks (expected as the legend needs space) but also the entire figure gets resized in a strange way (unexpected).

If I disable `tight_layout()` in `MATPLOTLIBPlot.show_default()` or remove the `bbox_to_anchor` parameter the issues disappear. 

How would you recommend to handle this issue?

Reply:
If you disable `tight_layout` in `MATPLOTLIBPlot.show_default()` and keep `bbox_to_anchor`, after the plot get's generated, if you call `plt.tight_layout()`, does it still cause strange resizing?

Reply:
No this did not resolve this issue on its own. Calling `plt.tight_layout()` after the fact seemed to not have any effect. Adjusting the anchoring of the `bbox`  seems to result in nice plots (and should be more stable from what I can tell):
![image](https://github.com/user-attachments/assets/c0af837d-ec6e-418d-9899-5769ae08fe1d)